transcom/mymove
# B 21423 int

## [B-21423](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21423)

## Summary

This PR does a few things in parallel for setting up Alaska/international SIT. It currently is not restricted by feature flags as none of the main functionality should be possible unless performed on an international shipment, which should be behind features flags itself.

This PR firstly removes county as a required field in the address model, instead making it optional to align it with our database setup. This was done as upstream AK work had issues with counties when development on this task first started. Related to this change, there will be two commits on the int branch not found on the main branch. They are: 
[Resolve int-only gen code error 42cdd41](https://github.com/transcom/mymove/pull/14079/commits/42cdd414c4d2a2f079fb7b7caebcc119390c8472) which is ineffectual as these changes were generated code changes made on the main branch under 24610ff, and 
[Resolve conflict with optional address 1 404c0ef](https://github.com/transcom/mymove/pull/14079/commits/404c0ef2059ff6a2270f08da37da078da888b39e), which resolves a merge conflict on INT with a test introduced in [B-21162](https://github.com/transcom/mymove/pull/13846/). This change will have to be made on the main branch before merged to main.

The goal of this task is to bring the International SIT codes up to par with the Domestic SIT codes for their creation and general rules (such as no IOSFSC without IOFSIT). It does not aim to allow for the updating of these items (though it does enable the prime sim pages for editing them), nor does it attempt to configure their pricing. These two features will be included in downstreams B-21424 and B-21425, respectively. This is why in some MTOServiceItemCreator service tests, the creator is still initialized with domestic pricing.

Per PO for implementation specifics, International SIT items are created whenever the Prime creates an Origin or Destination SIT on any international HHG Shipment. This is opposed to checking SIT addresses for oconus values, so the backlog and 1st story feature ACs are inaccurate.

In order to bring the international codes to par with the domestic codes, a migration had to be added to introduce IDSFSC and IOSFSC codes to the re_services table. These were late additions to the feature as they were unplanned in the original backlog breakdown, and didn't yet exist in milmove. No service_item_params were created for this service item category as these fall better under B-21425's mandate. This is the change that results in the database label and need for database team approval.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

Ensure that the enable Alaska feature flag is set to true!

1. Access the application as a Prime Sim user.
2. Find an appropriate HHG move with no SIT items.
3. Open your database, find the shipment you will use and set its market code to "i".
4. Alternatively, if you cannot do the above, create a new customer and make a new HHG Shipment move, and use it for testing.
5. Once you have your international shipment, create an origin SIT service item.
6. Additionally, create a destination SIT service item.
7. Ensure both populate under the shipment details, along with their automatically-created counterparts (should be the exact equivalent code of domestic service items created, but international starting with I).
8. You should be able to view the edit page, but you may not be able to update them. Again, this will come in a downstream task.
9. Log in as an office user (such as TOO) and view the move task order tab. 
10. Ensure that the service items you've created populate. You should be able to approve and deny them, though this isn't required.
11. This is the minimum testing to ensure AC is met. However, please test the above using swagger/postman if possible to ensure errors are correctly handled. 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
